### ISSUE TITLE
❓🐞@microsoft/sp-client-preview Missing in dependency chain (extra work req. for PNPM usage)

### ISSUE THREAD

**Issue by andrewconnell** [SharePoint/sp-dev-docs#2043]
#### Category
- [x] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

Running `pnpm install` should get all dependencies so when building the project, it builds with no errors when importing the **@microsoft/sp-client-preview** package.

#### Observed Behavior

When you run `pnpm install`, you don't get the **@microsoft/sp-client-preview** package. To get it, you must run `pnpm install --shamefully-flatten`.

Ref: https://pnpm.js.org/docs/en/faq.html#pnpm-does-not-work-with-your-project-here

#### Steps to Reproduce

* create a new project with `--plusbeta`.
* add the following import statement to the web part:

    ```ts
    import { MSGraphClient } from `@microsoft\sp-client-preview`
    ```

* add the following statement to the web part's `render()` method:

    ```ts
    const client: MSGraphClient = this.context.serviceScope.consume(MSGraphClient.serviceKey);
    ```

* run `pnpm install`
* run `gulp build`
* observe build errors
* use [solution 2 from the PNPM FAQ](https://pnpm.js.org/docs/en/faq.html#pnpm-does-not-work-with-your-project-here): run `pnpm install --shamefully-flatten`

  > add `--force` to rebuild the **node_modules** folder without having to delete it

* run `gulp build`
* observe no build errors

**Comment by andrewconnell** [comment]
Bit more info... I've tried to figure out where this should be correctly referenced as a dependency... but I'm not sure. Maybe in the root `package.json` of the project? Not sure... MSFT engineering would have to state the intention here...

**Comment by patmill** [comment]
Hey Andrew - this might be a silly questin, but is there a reference to @microsoft/sp-client-preview 1.5.0-plusbeta in your package.json file?

**Comment by andrewconnell** [comment]
Hey @patmill ... yes... the project was created with the `--plusbeta` flag and verified by looking at the **package.json**

**Comment by andrewconnell** [comment]
Confirmed this is still an issue in v1.5.1

**Comment by baywet** [comment]
still an issue in 1.6.0

**Comment by andrewconnell** [comment]
Appears resolved in v1.7.0, but a new related issue cropped up in ALL projects (not just `-plusbeta`): #2916
